src/server/kube-store.js

```diff
@@ -128,7 +138,89 @@ class KubeStore {
                 return;
             }
             default: {
-                throw new Error(`Could not patch noobaa, got ${JSON.stringify(body)}`);
+                throw new Error(`Could not patch noobaa, (status code: ${status_code}) got ${JSON.stringify(body)}`);
+            }
+        }
+    }
+
+    async create_backingstore(new_store) {
+        await this._init();
+        const path = get_backingstores_path(this._k8s_namespace);
+        const { status_code, body } = await this._make_k8s_api_request('POST', path, new_store);
+        switch (status_code) {
+            case 201: {
+                return body;
+            }
+            case 404: {
+                return null;
+            }
+            default: {
+                throw new Error(`Could not create backingstore, (status code: ${status_code}) got ${JSON.stringify(body)}`);
+            }
+        }
+    }
+
+    async delete_backingstore(name) {
+        await this._init();
+        const path = get_backingstores_path(this._k8s_namespace) + `/${name}`;
+        const { status_code, body } = await this._make_k8s_api_request('DELETE', path);
+        switch (status_code) {
+            case 200: {
+                return body;
+            }
+            case 404: {
+                return null;
+            }
+            default: {
+                throw new Error(`Could not delete backingstore, (status code: ${status_code}) got ${JSON.stringify(body)}`);
+            }
+        }
+    }
+
+    async read_backingstore(name) {
+        await this._init();
+        const path = get_backingstores_path(this._k8s_namespace) + `/${name}`;
+        const { status_code, body } = await this._make_k8s_api_request('GET', path);
+        switch (status_code) {
+            case 200: {
+                return body;
+            }
+            case 404: {
+                return null;
+            }
+            default: {
+                throw new Error(`Could not retrive backingstore, (status code: ${status_code}) got ${JSON.stringify(body)}`);
+            }
+        }
+    }
+
+    async patch_backingstore(name, patch) {
+        await this._init();
+        const path = get_backingstores_path(this._k8s_namespace) + `/${name}`;
+        const { status_code, body } = await this._make_k8s_api_request('PATCH', path, patch);
+        switch (status_code) {
+            case 200: {
+                return;
+            }
+            default: {
```

Comment:
I think you should take care of 404 in this case, in the case that the basking store you are trying to patch do not exists. So returning a bool make more sense.